src/main/webapp/dashboard-manager.js

```diff
@@ -33,65 +38,176 @@ class DashboardManager {
+    }));
+
+    this.table.setDataTable(this.data);
+
+    // Setup pie chart
+    removeAllChildren(this.pieChart);
+    this.configurePieChart(this.pieChart, this.data, selectorState, 1);
+  }
+
+  /* Setup data for standard table view */
+  async updateNormal() {
+    // TODO: use real data
+    this.data = this.initData();
+    this.table.setDataTable(this.data);
+  }
+
+  /* Create a (sub)PieChart with the appropriate data and event handlers */
+  configurePieChart(pieChart, baseData, selectorState, depth, parent) {
+    // Filer relevant entries from baseData based on which slice (if any) was selected
+    let filtered = filterDataFromParent(baseData, depth, parent);
+
+    // Perform the aggregation and set the result as the pieChart's data
+    let result = google.visualization.data.group(
+        filtered,
+        [depth - 1],
+        [{'column': filtered.getNumberOfColumns() - 1, 'aggregation': google.visualization.data.sum, 'type': 'number'}]);
+    pieChart.setView({'columns': [0, 1]});
+    pieChart.setDataTable(result);
+
+    if (isLastAggregation(selectorState.length, depth)) {
+      // We only want one event listener at a time, so we must remove/overwrite the previous one.
+      addOverwriteableChartEvent(
+        pieChart,
+        'select',
+        this.onSliceSelect.bind(this, pieChart, filtered, selectorState, depth + 1));
+    } else {
+      // TODO: Allow bulk updating the selected slice
+      addOverwriteableChartEvent(
+        pieChart,
+        'select',
+        () => {console.log("Chart slice was clicked!")});
+    }
+  }
+
+  onSliceSelect(parent, baseData, selectorState, depth) {
+    // Remove all current sub pie charts because a new slice has been selected and
+    // new sub pie charts will be generated
+    removeAllChildren(parent);
+
+    this.createSubPieChart(parent, baseData, selectorState, depth);
+  }
+
+  createSubPieChart(parent, baseData, selectorState, depth) {
+    const id = 'chart-' + depth;
+    let pieChart = createNewDivWithPieChart(id);
+
+    this.configurePieChart(pieChart, baseData, selectorState, depth, parent);
+    parent.childChart = pieChart;
+
+    this.draw();
+  }
+
   isAggregating() {
     // If no aggregation tags are selected, we are not currently aggregating
     return this.aggregationSelector.getState().selectedValues.length != 0;
   }
 
   draw() {
-    this.dashboard.draw(this.data);
-
-    this.table.setDataTable(this.data);
+    if (!this.drawnControls) {
+      this.aggregationSelector.draw();
+      this.drawnControls = true;
+    }
 
     this.table.draw();
-    this.aggregationSelector.draw();
-
     if (this.isAggregating()) {
-      this.pieChart.setDataTable(this.data);
       this.pieChart.draw();
+      this.drawChildren(this.pieChart);
+    }
+  }
+
+  drawChildren(chart) {
+    while (chart.childChart != null) {
+      chart = chart.childChart;
+      chart.draw();
     }
   }
 };
 
+function isLastAggregation(aggregationsDesired, aggregationsDone) {
```

Comment:
Shouldn't it be isNotLastAggregation?

Reply:
Nice catch, instead of renaming it though I inverted the logic such that it does actually check if it is the last aggregation.